src/network/protocol.rs

```diff
@@ -330,6 +329,50 @@ pub struct FixedIp {
     pub subnet_id: String,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Default, Ord, PartialOrd, Hash)]
```

Comment:
The CI fails because of missing Display (which I think should be the same as to_string())

Reply:
I will add `impl Display`.